File: split/newmath.pas
function newmath(w: scaled; s: smallnumber): halfword;
var
  p: halfword;
begin
  p := getnode(2);
  mem[p].hh.b0 := 9;
  mem[p].hh.b1 := s;
  mem[p + 1].int := w;
  newmath := p;
end;

File: cpp/newmath.cpp
#include "newmath.h"
#include "getnode.h"

halfword newmath(scaled w, smallnumber s)
{ 
	auto p = getnode(2);
	type(p) = math_node;
	subtype(p) = s;
	mem[p+1].int_ = w;
	return p;
}

